domain/src/main/kotlin/app/discord/user/repository/UserRepository.kt

```diff
@@ -1,13 +1,13 @@
 package app.discord.user.repository
 
-import app.discord.user.dto.UserRegisterEvent
+import app.discord.user.dto.UserIdentifier
 import app.discord.user.entity.User
 
+
 interface UserRepository {
-    fun findAllUsers(): List<User>
-    fun findUsersByUserId(userId: String): List<User>
-    fun findUsersByGuildId(guildId: String): List<User>
-    fun findUser(guildId: String, userId: String): User
-    fun registerUser(user: UserRegisterEvent)
+
+    fun findUser(userIdentifier: UserIdentifier): User?
+    fun findUserWithNullException(userIdentifier: UserIdentifier): User
+    fun insertUser(user: User)
     fun updateUser(user: User)
```

Comment:
```suggestion
    fun insertUser(user: User): User
    fun updateUser(user: User): User
```
업데이트 혹은 저장하고 난 뒤에 업데이트 된 객체를 Return 하는것이 일반적입니다.
돌려주지 않으면 변경된 User 정보를 알 수가 없습니다.